app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -1457,16 +1459,24 @@ public void run() {
               narrowedList = null;
               getSupportFragmentManager().popBackStack(NARROW,
                     FragmentManager.POP_BACK_STACK_INCLUSIVE);
+            clearSearch();
         }
     }
 
+    public void clearSearch()
+    {
+        searchView.setIconified(true);
```

Comment:
Nice! Here maybe a comment is in need to explain that the first call sets empty string in searchview textview and second call hides keyboard. Also, it be helpful if you can use JavaDoc comments for newly introduced functions. Once `.travis.yml` is merged we may set the severity of missing JavaDoc comments to error just to enhance code readability.